flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/adaptive/allocator/AbstractRequestSlotMatchingStrategyTest.java

```diff
@@ -0,0 +1,164 @@
+
+    @Test
+    void testMatchRequestsWithSlots() {
+        Collection<SlotAssignment> slotAssignments =
+                requestSlotMatchingStrategy.matchRequestsWithSlots(requestedGroups, freeSlots);
+        assertAssignments(slotAssignments);
+    }
+
+    protected static @Nonnull Map<TaskManagerLocation, Set<SlotAssignment>>
+            getAssignmentsPerTaskManager(Collection<SlotAssignment> assignments) {
+        return assignments.stream()
+                .collect(
+                        Collectors.groupingBy(
+                                assignment -> assignment.getSlotInfo().getTaskManagerLocation(),
+                                Collectors.toSet()));
+    }
+
+    private static TestingSlot createAnySlotOf(TaskManagerLocation tml) {
+        return new TestingSlot(new AllocationID(), ResourceProfile.ANY, tml);
+    }
+
+    private static ExecutionSlotSharingGroup createGroup(int executionVertices) {
+        return new ExecutionSlotSharingGroup(
+                IntStream.range(0, executionVertices)
+                        .mapToObj(ignored -> new ExecutionVertexID(new JobVertexID(), 0))
+                        .collect(Collectors.toSet()));
+    }
+}
+
+/** Test for {@link TasksBalancedRequestSlotMatchingStrategy}. */
+class TasksBalancedRequestSlotMatchingStrategyTest extends AbstractRequestSlotMatchingStrategyTest {
+
+    @Override
+    protected RequestSlotMatchingStrategy createRequestSlotMatcher() {
+        return TasksBalancedRequestSlotMatchingStrategy.INSTANCE;
+    }
+
+    @Override
+    protected void assertAssignments(Collection<SlotAssignment> assignments) {
+        Map<TaskManagerLocation, Set<SlotAssignment>> assignmentsPerTm =
+                getAssignmentsPerTaskManager(assignments);
+        assertThat(assignmentsPerTm)
+                .allSatisfy(
+                        (taskManagerLocation, slotAssignments) -> {
+                            assertThat(
+                                            slotAssignments.stream()
+                                                    .map(
+                                                            s ->
+                                                                    s.getTargetAs(
+                                                                                    ExecutionSlotSharingGroup
+                                                                                            .class)
+                                                                            .getLoading())
+                                                    .reduce(
+                                                            DefaultLoadingWeight.EMPTY,
+                                                            LoadingWeight::merge)
+                                                    .getLoading())
+                                    .isGreaterThanOrEqualTo(9f);
+                        });
+    }
+}
```

Comment:
It's a little bit strange to have this test case in an abstract class. Could we move it to a file with the same name?